connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -361,6 +373,16 @@ protected void handleLostAssignments(ConnectorsAndTasks lostAssignments,
         log.debug("Found the following connectors and tasks missing from previous assignments: "
                 + lostAssignments);
 
+        if (previousMembers.size() == memberConfigs.size() && scheduledRebalance <= 0) {
```

Comment:
Is it enough to trust that the # of workers has not changed, or should we compare the members, via something like:
```
if (previousMembers.equals(memberConfigs.keySet()) && scheduledRebalance <= 0) {
```
IOW, what happens if one worker disappeared about the same time that an operator added a new worker?

IIUC from the integration tests, this logic actually doesn't care which of these is the case -- all of the task assignments that were lost will be reassigned anyway, so it doesn't matter if the worker that gets those new assignments is the old worker that came back or a new worker. Is that right?

Reply:
I like the idea of checking the member configs. Could probably allow us to avoid duplicate tasks, even in this rare scenario of replacement of a departing node within the rebalance round itself.